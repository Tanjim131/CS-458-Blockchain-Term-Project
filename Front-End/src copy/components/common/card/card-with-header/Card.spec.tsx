import React from 'react';
import { shallow } from "enzyme";
import WalletCard from "./Card";
import { shallowToJson } from "enzyme-to-json";

describe('Wallet Component', () => {
    let wrapper: any;
    const mockFunc = jest.fn();

    beforeAll(() => {
        wrapper = shallow(
            <WalletCard
                icon=""
                title="title"
                coinLeft={10}
                headerColorClass="color"
                showButton={true}
                handleAddFreeCoin={mockFunc}
                showViewDetails={false}
                handleViewDetails={mockFunc}
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })

    it('Check the component when the showView Details is truthy', () => {
        wrapper.setProps({ showViewDetails: true });
        expect(wrapper).toBeTruthy();
    })
})