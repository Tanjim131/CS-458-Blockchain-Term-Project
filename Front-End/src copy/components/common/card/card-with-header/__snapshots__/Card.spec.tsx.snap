// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Component Check if component exists or not and capture snapshot 1`] = `
<Card
  className="my-wallet-color-card color"
  title="title"
>
  <div
    className="card-coin-and-button"
  >
    <div
      className="card-content"
    >
      <div
        className="coin-balance"
      >
        <FormattedMessage
          id="coin_balance"
          values={Object {}}
        />
      </div>
      <div
        className="coin-balance-amount"
      >
        10
      </div>
      <img
        className="coins-icon"
        src=""
      />
    </div>
    <Button
      className="add-free-coins-button"
      onClick={[MockFunction]}
      type={0}
    >
      <FormattedMessage
        id="add_free_coins"
        values={Object {}}
      />
    </Button>
  </div>
</Card>
`;
